python/spider-py/src/spider_py/client/task.py

```diff
@@ -0,0 +1,101 @@
+"""Spider client task module."""
+
+import inspect
+from collections.abc import Callable
+from types import FunctionType, GenericAlias
+from typing import get_args, get_origin
+
+from spider_py import core
+from spider_py.client.data import Data
+from spider_py.core import TaskInput, TaskOutput, TaskOutputData, TaskOutputValue
+from spider_py.type import to_tdl_type_str
+
+
+class TaskContext:
+    """Spider task context."""
+
+    # TODO: Implement task context for use in task executor
+
+
+# Check the TaskFunction signature at runtime.
+# Enforcing static check for first argument requires the use of Protocol. However, functions, which
+# are Callable, are not considered a Protocol without explicit cast.
+TaskFunction = Callable[..., object]
+
+
+def _is_tuple(t: type | GenericAlias) -> bool:
+    """
+    :param t:
+    :return: Whether t is a tuple.
+    """
+    return get_origin(t) is tuple
+
+
+def _process_parameters(task: core.Task, signature: inspect.Signature) -> None:
+    """
+    Checks the parameters validity and add them to the task.
+    :param task:
+    :param signature:
+    :raises TypeError: If the parameters are invalid.
+    """
+    params = list(signature.parameters.values())
+    if not params or params[0].annotation is not TaskContext:
+        msg = "First argument is not a TaskContext."
+        raise TypeError(msg)
+    for param in params[1:]:
+        if param.kind in {inspect.Parameter.VAR_POSITIONAL, inspect.Parameter.VAR_KEYWORD}:
+            msg = "Variadic parameters are not supported."
+            raise TypeError(msg)
+        if param.annotation is inspect.Parameter.empty:
+            msg = "Argument must have type annotation"
```

Comment:
```suggestion
            msg = "Parameters must have type annotation."
```
It is "parameter", right?
Also the period is missing.